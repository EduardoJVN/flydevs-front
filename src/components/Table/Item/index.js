import react , {Fragment} from 'react'
import {Button,Card, TextField, Container, Box, CircularProgress , Grid} from '@material-ui/core';
import Typography from '@material-ui/core/Typography';
export default function Item ({pokemons}){
    return(
        <Fragment>
            {pokemons.map((data, index) => (
                <Grid item xs={12} className="itemList" key={index}>
                    <Box border={1}>
                        <Grid container>
                            <Grid item xs={4} sm={4} md={2} lg={2} xl={2}>
                                <img style={{ width: '100%', height: 150 }}  src={data.image} className="image" />
                            </Grid>
                            <Grid item xs={8} sm={8} md={10} lg={10} xl={10}>
                                <Typography component="h5" variant="h5">
                                    {data.name}
                                </Typography>
                            </Grid>
                        </Grid>
                    </Box>
                </Grid>
            ))}
        </Fragment>
        
    )
}