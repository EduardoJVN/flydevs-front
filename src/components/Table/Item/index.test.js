import React from 'react';
import Item from './index';
import ReactDOM from 'react-dom';
import Enzyme , {shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
Enzyme.configure({ adapter: new Adapter() });

it('renders correctly', () => {
    const div = document.createElement('div');
    const pokemons = [
        {name:"pikachu",image:"https://static.wikia.nocookie.net/pokemon-fano/images/6/6f/Poke_Ball.png/revision/latest/scale-to-width-down/340?cb=20140520015336"}
    ]
    ReactDOM.render(<Item pokemons={pokemons}/>,div);
    expect(div).toMatchSnapshot();
});

// it('renders item correctly', () => {
//     const pokemons = [
//         {name:"pikachu",image:"https://static.wikia.nocookie.net/pokemon-fano/images/6/6f/Poke_Ball.png/revision/latest/scale-to-width-down/340?cb=20140520015336"}
//     ]
//     const wrapper = shallow(<Item pokemons={pokemons}/>);
//     expect(wrapper.find('.itemList').length).toBe(1)
// });