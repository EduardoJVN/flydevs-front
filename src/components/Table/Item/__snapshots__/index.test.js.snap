// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root itemList MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-sm-4 MuiGrid-grid-md-2 MuiGrid-grid-lg-2 MuiGrid-grid-xl-2"
        >
          <img
            class="image"
            src="https://static.wikia.nocookie.net/pokemon-fano/images/6/6f/Poke_Ball.png/revision/latest/scale-to-width-down/340?cb=20140520015336"
            style="width: 100%; height: 150px;"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-sm-8 MuiGrid-grid-md-10 MuiGrid-grid-lg-10 MuiGrid-grid-xl-10"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5"
          >
            pikachu
          </h5>
        </div>
      </div>
    </div>
  </div>
</div>
`;
